File: Baekjoon/문자열/2941.py
# https://www.acmicpc.net/problem/2941

import sys
input = sys.stdin.readline

s = input().rstrip()
cnt = 0
i = 0

while i < len(s):
    if s[i] == 'c':
        if i + 1 < len(s) and (s[i + 1] == '=' or s[i + 1] == '-'):
            i += 1
    elif s[i] == 'd':
        if i + 1 < len(s) and s[i + 1] == '-':
            i += 1
        elif i + 2 < len(s) and s[i + 1] == 'z' and s[i + 2] == '=':
            i += 2
    elif s[i] == 'l' or s[i] == 'n':
        if i + 1 < len(s) and s[i + 1] == 'j':
            i += 1
    elif s[i] == 's' or s[i] == 'z':
        if i + 1 < len(s) and s[i + 1] == '=':
            i += 1
    cnt += 1
    i += 1

print(cnt)

File: Baekjoon/문자열/2941.cpp
// https://www.acmicpc.net/problem/2941

#include <iostream>
#include <string>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    string s;
    cin >> s;

    int cnt = 0;
    for (int i = 0; i < s.size(); i++) {
        if (s[i] == 'c' && (s[i + 1] == '=' || s[i + 1] == '-')) {
            i++;
        } else if (s[i] == 'd' && s[i + 1] == 'z' && s[i + 2] == '=') {
            i += 2;
        } else if ((s[i] == 'd' && s[i + 1] == '-') || (s[i] == 'l' && s[i + 1] == 'j') || (s[i] == 'n' && s[i + 1] == 'j') || (s[i] == 's' && s[i + 1] == '=') || (s[i] == 'z' && s[i + 1] == '=')) {
            i++;
        }
        cnt++;
    }

    cout << cnt << "\n";

    return 0;
}